pyteomics/mztab.py

```diff
@@ -744,10 +744,13 @@ def _parse(self):
                 self.small_molecule_evidence_table.add(tokens[1:])
 
     def _determine_schema_version(self):
-        version_parsed, variant = re.search(r"(?P<schema_version>\d+.\d+.\d+)(?:-(?P<schema_variant>[MP]))?", self.version).groups()
+        version_parsed, variant = re.search(r"(?P<schema_version>\d+(?:.\d+(?:.\d+)?)?)(?:-(?P<schema_variant>[MP]))?", str(self.version)).groups()

```

Comment:
Quick question: are we matching any character `.` instead of a literal `\.` here? Should we fortify the pattern?

```suggestion
        version_parsed, variant = re.search(r"(?P<schema_version>\d+(?:\.\d+(?:\.\d+)?)?)(?:-(?P<schema_variant>[MP]))?", str(self.version)).groups()
```
Doesn't matter all that much, because the splitting occurs later.

Reply:
You're correct, we should be matching `\.` not `.`. It was a flaw in my original pattern too. If a really malformed version string were tested it would fail at a different location instead of failing to match the regex.